src/test/java/se/sundsvall/contract/integration/db/model/generator/ContractIdGeneratorTest.java

```diff
@@ -0,0 +1,93 @@
+package se.sundsvall.contract.integration.db.model.generator;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase.Replace.NONE;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+import org.hibernate.engine.jdbc.spi.JdbcCoordinator;
+import org.hibernate.engine.jdbc.spi.StatementPreparer;
+import org.hibernate.engine.spi.SharedSessionContractImplementor;
+import org.hibernate.generator.EventType;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.test.context.ActiveProfiles;
+
+import jakarta.persistence.PersistenceException;
+
+@DataJpaTest
+@AutoConfigureTestDatabase(replace = NONE)
+@ActiveProfiles("junit")
+class ContractIdGeneratorTest {
+
+	@Mock
+	private SharedSessionContractImplementor session;
+
+	@Mock
+	private Object currentValue;
+
+	@Mock
+	private StatementPreparer mockStatementPreparer;
+
+	@Mock
+	private JdbcCoordinator mockJdbcCoordinator;
+
+	@Mock
+	private ResultSet mockResultSet;
+
+	@Mock
+	private PreparedStatement mockPreparedStatement;
+
+	private static final String GENERATE_ID_QUERY = """
+        SELECT CONCAT(YEAR(CURRENT_DATE), '-', LPAD(NEXT VALUE FOR `contract_id_seq`, 5, 0))
+        """;
+
+	private ContractIdGenerator contractIdGenerator;
+
+	@BeforeEach
+	public void setUp() throws SQLException {
+		contractIdGenerator = new ContractIdGenerator();
```

Comment:
Use the `@InjectMocks` instead?

```
@InjectMocks
private ContractIdGenerator contractIdGenerator;
```

Reply:
Got lost while fiddling with the test, will reintroduce.